aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+]
+REQUIRED_CZI_CHUNK_DIMS = [
+    dim if dim != DimensionNames.Samples else "A" for dim in REQUIRED_CHUNK_DIMS
+]
+DEFAULT_CZI_DIMENSION_ORDER_LIST = [
+    dim if dim != DimensionNames.Samples else "A"
+    for dim in DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES
+]
+DEFAULT_CZI_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES = [
+    dim if dim != DimensionNames.Samples else "A"
+    for dim in DEFAULT_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES_AND_SAMPLES
+]
+
+
+###############################################################################
+
+
+class CziReader(Reader):
+    """
+    Wraps the aicspylibczi API to provide the same aicsimageio Reader API but for
+    volumetric Zeiss CZI images.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    chunk_by_dims: Union[str, List[str]]
+        Which dimensions to create chunks for.
+        Default: DEFAULT_CHUNK_DIMS
+        Note: Dimensions.SpatialY, Dimensions.SpatialX, and DimensionNames.Samples,
+        will always be added to the list if not present during dask array
+        construction.
+
+    Notes
+    -----
+    To use this reader, install with: `pip install aicsimageio[czi]`.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            with fs.open(path) as open_resource:
+                CziFile(open_resource)
+                return True
+
+        except RuntimeError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        chunk_by_dims: Union[str, List[str]] = DEFAULT_CHUNK_DIMS,
+    ):
+        # doesn't do anything in this case but here for completeness
+        super(CziReader, self).__init__(image=image)
+
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+
+        # Store params
+        if isinstance(chunk_by_dims, str):
+            chunk_by_dims = list(chunk_by_dims)
+
+        self.chunk_by_dims = chunk_by_dims
+
+        # Delayed storage
+        self._px_sizes: Optional[types.PhysicalPixelSizes] = None
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            with self._fs.open(self._path) as open_resource:
+                czi = CziFile(open_resource)
+                xpath_str = "./Metadata/Information/Image/Dimensions/S/Scenes/Scene"
+                meta_scenes = czi.meta.findall(xpath_str)
+                scene_names = [x.get("Name") for x in meta_scenes]
+                # if the scene is implicit just assign it name Scene:0
+                if len(scene_names) < 1:
+                    scene_names = ["Scene:0"]
```

Comment:
The result of calling `generate_ome_image_id` would be Image:0 yes. But I am saying to explicitly use that function so that if the OME standard changes we can just update in a single place.